tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js

```diff
@@ -349,6 +356,62 @@ class LifecycleUtility extends ReplicationUtility {
             return cb(null, res.Uploads.length > 0);
         });
     }
+
+    /**
+     * Put an object restore request
+     *
+     * @param {string} [versionId] - version id of an object
+     * @param {function} cb - callback function
+     *
+     * @returns {undefined} undefined
+     */
+    putRestoreObject(versionId, cb) {
+        const params = {
+            Bucket: this.bucket,
+            Key: this.key,
+            RestoreRequest: {
+                Days: 1,
+                Tier: 'Standard',
+            },
+        };
+        if (versionId) {
+            params.VersionId = versionId;
+        }
+        this.s3.restoreObject(params, cb);
+    }
+
+    /**
+     * Wait until an archived object is restored
+     *
+     * @param {string} [versionId] - version id of an object
+     * @param {function} cb - callback function
+     *
+     * @returns {undefined} undefined
+     */
+    waitUntilRestored(versionId, cb) {
+        let shouldContinue;
+        const params = {
+            Bucket: this.bucket,
+            Key: this.key,
+        };
+        if (versionId) {
+            params.VersionId = versionId;
+        }
+        return async.doWhilst(
+            next => this.s3.headObject(params, (err, data) => {
+                if (err) {
+                    return next(err);
+                }
+                shouldContinue = !data.StorageClass;
```

Comment:
not sure about this check to verify restoration, because `StorageClass` is removed or not part of the metadata after restoration.

Reply:
Since this function is called: `waitUntilRestored`. Its job is to wait until the object is restored. Checking that the storage class change from cold to STANDARD (empty) should be sufficient to assess that the object has been retored. If additional checks on the object metadata are needed to verify restore, we could pass the `headObject` data (or make other calls) to the function callback and add additional checks once `waitUntilRestored` is done. https://github.com/scality/Zenko/pull/1693/files#diff-efaa77cd5fb4679d64cdd85814569d9adef4771eb2a7aba1752ca3a8e39778d2R47

Reply:
I think I added this comment before your reference to AWS behavior towards object's metadata when restored (restoring to STANDARD would remove `StorageClass`, when `headObject` is issued).